### PR TITLE
Add integration test for defaulting logic

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/11656

### CODE REVIEW

File: integration/test/mutation/main_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+	{
+		values := `Installation:
+  V1:
+    Registry:
+      Domain: quay.io`
+		apps := []apptest.App{
+			{
+				CatalogName:   prodCatalogName,
+				Name:          "cert-manager-app",
```
Comment (tomahawk28) [inline integration/test/mutation/main_test.go:60]: Sorry for my ignorance, but why we need `cert-manager-app` for app-admission-controller?

File: integration/test/mutation/main_test.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			KubeConfigPath: env.KubeConfig(),
+			Logger:         logger,
+		})
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	{
+		values := `Installation:
```
Comment (tomahawk28) [inline integration/test/mutation/main_test.go:53]: Can we put this `values` in `intergration/template/values.go`? I think many repos are following that pattern already.

File: integration/test/mutation/main_test.go (hunk 1 above)
Comment (rossf7) [reply integration/test/mutation/main_test.go:60]: The webhooks have to be served over TLS. So we use our cert-manager to issue a self signed cert.

File: integration/test/mutation/main_test.go (hunk 2 above)
Comment (rossf7) [reply integration/test/mutation/main_test.go:53]: Good call. I'll move this.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-- Add validation webhook for app CRs.
+- Add mutation webhook with defaulting logic that is enabled for app CRs with
+`app-operator.giantswarm.io/version` label value >= `3.0.0`.
+- Add validation webhook that is enabled for app CRs with
+`app-operator.giantswarm.io/version` label value >= `3.0.0`.
```
Comment (rossf7) [inline CHANGELOG.md:17]: Going to create a 0.1.0 release after this for testing. Will add pushing to app collections later.

File: integration/test/mutation/main_test.go (hunk 1 above)
Comment (rossf7) [reply integration/test/mutation/main_test.go:59]: I added a comment for this. 👍

**Review by rossf7 — COMMENTED** [review]
@tomahawk28 Changes made. PTAL

File: integration/test/mutation/main_test.go (hunk 1 above)
Comment (tomahawk28) [reply integration/test/mutation/main_test.go:59]: Nice, Thanks!

**Review by tomahawk28 — APPROVED** [review]

**Pull request merged**